File: backend/app/api/v1/position.py
from fastapi import APIRouter, Depends, HTTPException, Query
from sqlalchemy.orm import Session
from typing import Optional
from app.database import get_db
from app.schemas.position import (
    PositionCreate, PositionUpdate, PositionResponse, PositionListResponse
)
from app.crud import position as crud

# 所有接口前缀；接口文档分组名
router = APIRouter(prefix="/api/v1/positions", tags=["岗位管理"])

@router.get("", response_model=PositionListResponse, summary="获取岗位列表")
def get_positions(
    page: int = Query(1, ge=1, description="页码"),
    page_size: int = Query(10, ge=1, le=100, description="每页条数"),
    position_name: Optional[str] = Query(None, description="岗位名称"),
    department: Optional[str] = Query(None, description="部门"),
    status: Optional[int] = Query(None, description="状态"),
    db: Session = Depends(get_db)
):
    """获取岗位列表，支持分页和筛选"""
    total, items = crud.get_position_list(
        db, page, page_size, position_name, department, status
    )
    return PositionListResponse(
        total=total,
        items=[PositionResponse.model_validate(item) for item in items],
        page=page,
        page_size=page_size
    )

@router.get("/{position_id}", response_model=PositionResponse, summary="获取岗位详情")
def get_position(position_id: int, db: Session = Depends(get_db)):
    """根据ID获取岗位详情"""
    db_position = crud.get_position_by_id(db, position_id)
    if not db_position:
        raise HTTPException(status_code=404, detail="岗位不存在")
    return PositionResponse.model_validate(db_position)

@router.post("", response_model=PositionResponse, summary="创建岗位")
def create_position(position: PositionCreate, db: Session = Depends(get_db)):
    """创建新岗位"""
    return PositionResponse.model_validate(crud.create_position(db, position))

@router.put("/{position_id}", response_model=PositionResponse, summary="更新岗位")
def update_position(
    position_id: int,
    position: PositionUpdate,
    db: Session = Depends(get_db)
):
    """更新岗位信息"""
    db_position = crud.update_position(db, position_id, position)
    if not db_position:
        raise HTTPException(status_code=404, detail="岗位不存在")
    return PositionResponse.model_validate(db_position)

@router.delete("/{position_id}", summary="删除岗位")
def delete_position(position_id: int, db: Session = Depends(get_db)):
    """删除岗位（软删除）"""
    db_position = crud.delete_position(db, position_id)
    if not db_position:
        raise HTTPException(status_code=404, detail="岗位不存在")
    return {"message": "删除成功"}

File: frontend/src/api/position.js
import request from "./request";

// 获取岗位列表
export function getPositionList(params) {
  return request.get("/positions", { params });
}

// 获取岗位详情

// 这是一个使用ES6模块语法的导出函数，命名为getPositionDetail
// 函数接受一个参数id，用于指定要查询的职位ID
// 函数内部使用request.get()方法发送一个GET请求到API端点/positions/${id}
// 这里使用了模板字符串（template literals）来动态构建API路径，将传入的id参数插入到URL中
// 函数返回这个HTTP请求的Promise对象
// 用途
// 这个函数通常用于前端应用程序中，当需要显示特定职位的详细信息时调用
// 它封装了获取职位数据的API调用，使代码更加模块化和可重用
// 通过导出这个函数，其他模块可以导入并使用它来获取职位数据
export function getPositionDetail(id) {
  return request.get(`/positions/${id}`);
}

// 创建岗位
export function createPosition(data) {
  return request.post("/positions", data);
}

// 更新岗位
export function updatePosition(id, data) {
  return request.put(`/positions/${id}`, data);
}

// 删除岗位
export function deletePosition(id) {
  return request.delete(`/positions/${id}`);
}